include/amici/defines.h

```diff
@@ -1,8 +1,18 @@
 #ifndef AMICI_DEFINES_H
 #define AMICI_DEFINES_H
+#include <cmath>
+
+/** MS definition of PI and other constants */
+#define _USE_MATH_DEFINES
+#ifndef M_PI
+/** define PI if we still have no definition */
+#define M_PI 3.14159265358979323846
+#endif
 
 namespace amici {
 
+constexpr double pi = M_PI;
```

Comment:
Can we merge that with the lines above and conditionally set `amici::pi` without introducing `M_PI` in the public header? (And consequently only use `amici::pi`)